### PR TITLE
Process lighthouse txs in the sequencer

### PR DESCRIPTION
**Pull request by wanglonghong**
## Type of change

- [ ] Docs change / dependency upgrade
- [ ] Configuration / tooling changes
- [x] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires changes in customer code

## Related Issue(s)

Fixes

### CODE REVIEW

File: packages/agents/sequencer/src/lib/operations/lighthouse.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  await cache.lighthousetxs.storeLightHouseData(lighthouseData);
+  logger.info("Created a lighthouse tx", requestContext, methodContext, { transferId, lighthouseData });
+
+  const message: Message = {
+    transferId: transfer.transferId,
+    originDomain: transfer.xparams!.originDomain,
+    type: MessageType.SlowPath,
+  };
+
+  await mqClient.publish(config.messageQueue.publisher!, {
```
Comment (rhlsthrm) [inline packages/agents/sequencer/src/lib/operations/lighthouse.ts:134]: Hmm, is this using the same message queue keys as the fast path? I want to make sure it's fully separated so the type/routingKey should be properly namespaced, so we should make it so it always publishes with a separate name.

File: packages/agents/sequencer/src/lib/operations/relayer.ts [hunk 2]
```diff
@@ -96,3 +105,81 @@ export const sendToRelayer = async (
   const taskId = await relayer.send(destinationChainId, destinationConnextAddress, encodedData, _requestContext);
   return taskId;
 };
+
+export const sendLightHouseDataToRelayer = async (
```
Comment (rhlsthrm) [inline packages/agents/sequencer/src/lib/operations/relayer.ts:109]: Let's break this file up into 2 files, `relayerFast` and `relayerSlow`.

File: packages/agents/sequencer/src/sequencer.ts [hunk 3]
```diff
@@ -122,7 +124,12 @@ export const execute = async (_configOverride?: SequencerConfig) => {
     context.adapters.contracts = getContractInterfaces();
     context.adapters.relayer = await setupRelayer();
 
-    await executeAuction(transferId, requestContext);
+    if (messageType == MessageType.Auction) {
+      await executeAuction(transferId, requestContext);
+    } else if (messageType == MessageType.SlowPath) {
```
Comment (rhlsthrm) [inline packages/agents/sequencer/src/sequencer.ts:129]: OK I see here these are namespaced, nice!

**Review by rhlsthrm — CHANGES_REQUESTED** [review]
This is great, nice work on this! Main things are around naming in a less confusing way. Here's what would help:
- Don't have anything named `auctions` etc, it should be `executeFast` and `executeSlow`.
- Split up all the logic into separate files i.e. `relayer.ts` should definitely be split.

File: packages/agents/lighthouse/src/lib/operations/execute.ts [hunk 4]
```diff
@@ -38,5 +35,5 @@ export const execute = async (args: ExecuteArgs, transferId: string): Promise<vo
   }
 
   const encodedData = contracts.connext.encodeFunctionData("execute", [args]);
-  await sendToRelayer(args, encodedData, transferId, requestContext);
+  await sendBidsToRelayer(args, encodedData, transferId, requestContext);
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/agents/lighthouse/src/lib/operations/execute.ts:38]: oh you're right

File: packages/agents/sequencer/src/lib/operations/lighthouse.ts (hunk 1 above)
Comment (wanglonghong) [reply packages/agents/sequencer/src/lib/operations/lighthouse.ts:134]: Yes. What I was thinking here is basically to make the message broker any kind of message, not just execute-fast ones. so we can integrate more like AMB off-chain stuff.

File: packages/agents/sequencer/src/lib/operations/auctions.ts [hunk 5]
```diff
@@ -42,10 +42,10 @@ export const storeBid = async (bid: Bid, _requestContext: RequestContext): Promi
   }
 
   // check if bid router version is compatible with hosted sequencer
-  const checkVersion = compare(bid.routerVersion, config.supportedBidVersion!, "<");
+  const checkVersion = compare(bid.routerVersion, config.supportedVersion!, "<");
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/agents/sequencer/src/lib/operations/auctions.ts:45]: Ok will go through the link you attached and update as needed

File: packages/agents/sequencer/src/sequencer.ts (hunk 3 above)
Comment (wanglonghong) [reply packages/agents/sequencer/src/sequencer.ts:129]: Nit: That would be better to update MessageType.Auction like MessageType.FastPath?

File: packages/agents/sequencer/src/lib/operations/lighthouse.ts [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+      });
+    }
+    // Store the transfer locally. We will use this as a reference later when we execute this transfer
+    // in the cycle, for both encoding data and passing relayer fee to the relayer.
+    await cache.transfers.storeTransfers([transfer]);
+  }
+
+  if (!transfer.destination?.reconcile || transfer.destination?.execute) {
+    // This transfer has already been Executed or not Reconciled yet, so slow liquidity is no longer valid.
+    throw new InvalidSlowLiqTransfer({
```
[the remark this answers is not in the document]
Comment (wanglonghong) [reply packages/agents/sequencer/src/lib/operations/lighthouse.ts:90]: It just checks whether its a slow transfer or not. we don't need to process fast transfer here

File: packages/agents/sequencer/src/sequencer.ts (hunk 3 above)
Comment (rhlsthrm) [reply packages/agents/sequencer/src/sequencer.ts]: Yes, correct this would be better!

**Pull request merged**